medical/models/medical_patient.py

```diff
@@ -1,105 +1,113 @@
-# Special Credit and Thanks to Thymbra Latinoamericana S.A.
-#
-# This program is free software: you can redistribute it and/or modify
-# it under the terms of the GNU Affero General Public License as
-# published by the Free Software Foundation, either version 3 of the
-# License, or (at your option) any later version.
-#
-# This program is distributed in the hope that it will be useful,
-# but WITHOUT ANY WARRANTY; without even the implied warranty of
-# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-# GNU Affero General Public License for more details.
-#
-# You should have received a copy of the GNU Affero General Public License
-#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
-#
-# #############################################################################
-from openerp import models, fields, api
+# Copyright 2004 Tech-Receptives
+# Copyright 2016 LasLabs Inc.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
 from datetime import datetime
-from openerp.tools.translate import _
 from dateutil.relativedelta import relativedelta
 
+from odoo import _, api, fields, models, tools
+from odoo.modules import get_module_resource
+from odoo.exceptions import ValidationError
+
 
 class MedicalPatient(models.Model):
-    '''
+    """
     The concept of Patient included in medical.
-    '''
+    """
     _name = 'medical.patient'
     _description = 'Medical Patient'
-    _inherits = {'res.partner': 'partner_id', }
+    _inherit = 'medical.abstract.entity'
+
+    age = fields.Char(
+        compute='_compute_age',
+    )
+    identification_code = fields.Char(
+        string='Internal Identification',
+        help='Patient Identifier provided by the Health Center.'
+             '(different from the Social Security Number)',
+    )
+    general_info = fields.Text(
+        string='General Information',
+    )
+    is_deceased = fields.Boolean(
+        compute='_compute_is_deceased',
+    )
+    marital_status = fields.Selection([
+        ('s', 'Single'),
+        ('m', 'Married'),
+        ('w', 'Widowed'),
+        ('d', 'Divorced'),
+        ('x', 'Separated'),
+        ('z', 'law marriage'),
+    ], )
+    is_pregnant = fields.Boolean(
+        help='Check this if the patient if pregnant',
+    )
+    date_death = fields.Datetime(
+        string='Deceased Date',
+    )
 
-    @api.one
+    @api.multi
     def _compute_age(self):
-        """
-        age computed depending of the birth date of the
-        membership request
+        """ Age computed depending based on the birth date in the
+         membership request.
         """
         now = datetime.now()
-        if self.dob:
-            dob = fields.Datetime.from_string(self.dob)
-
-            if self.deceased:
-                dod = fields.Datetime.from_string(self.dod)
-                delta = relativedelta(dod, dob)
-                deceased = _(' (deceased)')
+        for record in self:
```

Comment:
>  (I hate talking about that :/ )

It's sad, I think I've grown numb to this after being in medical so long. Thanks for making me realize 😛 

Perinatal mortality is measured in a different way from what I understand. There can't really be an instance of you dying before you're born, because if you weren't born how did you die? I believe those dates remain the same in this instance, although I'll bounce off a few experts to make sure.

The measurement for this is typically how far along in pregnancy the mother was (or in the instance of neonatal - up to 7 days after birth). We'll probably include this all in the pediatrics module.

That said, negative ages constraint 👍